cnxepub/formatters.py

```diff
@@ -317,15 +319,36 @@ def _fix_namespaces(html):
     return etree.tostring(let, pretty_print=True, encoding='utf-8')
 
 
-def exercise_callback_factory(match, url_template):
+def _replace_tex_math(node, mml_url):
+    """call mml-api service to replace TeX math in body of node with mathml"""
+
+    res = requests.post(mml_url, {'math': node.text,
+                                  'mathType': 'TeX',
+                                  'mml': 'true'})
+    if res:
+        eq = res.json()
+        mml = etree.fromstring(eq['components'][0]['source'])
+        if node.tag.endswith('span'):
+            mml.set('display', 'inline')
+        elif node.tag.endswith('div'):
+            mml.set('display', 'block')
+        return mml
+    else:
+        return None
```

Comment:
requests  Response object is special if 'OK' (basically 2XX or 3XX): 
```
>>> res=requests.get('http://example.com/nothing')
>>> res
<Response [404]>
>>> if res:
...   print True
... else:
...   print False
... 
False
>>> 
```
Code is here: https://github.com/kennethreitz/requests/blob/master/requests/models.py#L622-L628